### PR TITLE
NT-85 construction system

### PR DESCRIPTION
**Pull request by PatrickSlag**

### CODE REVIEW

**Review by jortdr — COMMENTED** [review]
Code looks fine, no remarks, still need to test it in Unity

File: Assets/Scripts/Presenters/BuildingPresenter.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Collections.Generic;
+using Enums;
+using Models;
+using UnityEngine;
+
+namespace Presenters
+{
+    public class BuildingPresenter : MonoBehaviour
```
Comment (VarioCode) [inline Assets/Scripts/Presenters/BuildingPresenter.cs:10]: I believe some of the logic should be moved to a service, please take a look at this.

**Review by VarioCode — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Presenters/BuildingPresenter.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                _buildStructures[tribe].Add(modelList[0]);
+                modelList[0].SetActive(true);
+            }
+
+            if (modelList.Count == 2)
+            {
+                _buildStructures[tribe].Add(modelList[1]);
+                modelList[0].SetActive(false);
+                modelList[1].SetActive(true);
+            }
```
Comment (jortdr) [inline Assets/Scripts/Presenters/BuildingPresenter.cs:121]: Add comments explaining what this does please

**Review by jortdr — APPROVED** [review]

File: Assets/Scripts/Presenters/BuildingPresenter.cs (hunk 2 above)
Comment (VarioCode) [reply Assets/Scripts/Presenters/BuildingPresenter.cs:121]: Yea, if you could add one please, after that its ready for merch imo

**Review by VarioCode — APPROVED** [review]

**Review by jortdr — APPROVED** [review]

**Pull request merged**